modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authorization/authorizer/rdbms/resource/AuthResourceStoreImpl.java

```diff
@@ -0,0 +1,192 @@
+import io.ballerina.messaging.broker.auth.BrokerAuthConfiguration;
+import io.ballerina.messaging.broker.auth.authorization.AuthProvider;
+import io.ballerina.messaging.broker.auth.authorization.AuthResourceStore;
+import io.ballerina.messaging.broker.auth.authorization.authorizer.rdbms.resource.dao.impl.AuthResourceDaoFactory;
+import io.ballerina.messaging.broker.auth.exception.BrokerAuthDuplicateException;
+import io.ballerina.messaging.broker.auth.exception.BrokerAuthException;
+import io.ballerina.messaging.broker.auth.exception.BrokerAuthNotFoundException;
+import io.ballerina.messaging.broker.auth.exception.BrokerAuthServerException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+import javax.sql.DataSource;
+
+/**
+ * Class provides implementation of @{@link AuthResourceStore} with database based auth resource store.
+ */
+public class AuthResourceStoreImpl implements AuthResourceStore {
+
+    /**
+     * User cache which contains authResource Key vs cache entry.
+     */
+    private LoadingCache<ResourceCacheKey, AuthResource> authResourceCache;
+
+    private AuthResourceDaoFactory authResourceDaoFactory;
+
+    private AuthProvider authProvider;
+
+    public AuthResourceStoreImpl(BrokerAuthConfiguration brokerAuthConfiguration, DataSource dataSource,
+                                 AuthProvider authProvider) {
+        this.authResourceDaoFactory = new AuthResourceDaoFactory(dataSource);
+        this.authProvider = authProvider;
+        this.authResourceCache = CacheBuilder.newBuilder()
+                                             .maximumSize(brokerAuthConfiguration.getAuthorization().getCache()
+                                                                                 .getSize())
+                                             .expireAfterWrite(brokerAuthConfiguration.getAuthorization()
+                                                                                      .getCache()
+                                                                                      .getTimeout(), TimeUnit.MINUTES)
+                                             .build(new AuthResourceCacheLoader());
+    }
+
+    @Override
+    public boolean authorize(String resourceType, String resourceName, String action, String userId,
+                             Set<String> userGroups) throws BrokerAuthServerException, BrokerAuthNotFoundException {
+        AuthResource authResource = read(resourceType, resourceName);
+        return Objects.nonNull(authResource) && (authResource.getOwner().equals(userId) ||
+                authResource.getActionsUserGroupsMap().get(action).stream()
+                            .filter(userGroups::contains)
+                            .findAny()
+                            .isPresent());
```

Comment:
We can replace this with `anyMatch(userGroups::contains));`

Reply:
Fixed in d43ea86